src/lib/markets/graph.ts

```diff
@@ -983,3 +983,140 @@ export const getPriceAtVersion = async ({
 
   return res.marketVersionPrice?.price ?? 0n
 }
+
+export type TradeHistoryItem<T> = {
+  side: string
+  delta: bigint
+  collateralDelta: bigint
+  priceImpactFee: bigint
+  fromOracleVersion: bigint
+  toOracleVersion: bigint
+  accumulatedValue: bigint
+  accumulatedPnl: bigint
+  accumulatedFunding: bigint
+  accumulatedInterest: bigint
+  accumulatedPositionFee: bigint
+  accumulatedKeeper: bigint
+  date: Date
+  transactionHash: string
+  accountPositionProcesseds: T
+}
+
+export async function fetchTradeHistory({
+  graphClient,
+  address,
+  first,
+  offset,
+}: {
+  graphClient: GraphQLClient
+  address: Address
+  first: number
+  offset: number
+}) {
+  const tradeHistoryQuery = gql(`
+   query fetchTradeHistory($account: Bytes!, $first: Int!, $offset: Int!) {
+    accountPositionProcesseds(
+      where: { account: $account },
+      orderBy: toOracleVersion,
+      orderDirection: desc,
+      first: $first,
+      skip: $offset,
+    ) {
+      account,
+      market,
+      fromOracleVersion,
+      toOracleVersion,
+      accumulatedValue,
+      accumulatedPnl,
+      accumulatedFunding,
+      accumulatedInterest,
+      accumulationResult_positionFee,
+      accumulationResult_keeper,
+      update {
+        version,
+        delta,
+        side,
+        collateral,
+        priceImpactFee,
+        blockTimestamp,
+        transactionHash
+      }
+    }
+   }
+  `)
+
+  const { accountPositionProcesseds } = await graphClient.request(tradeHistoryQuery, {
+    account: address,
+    first,
+    offset,
+  })
+
+  const entities = accountPositionProcesseds.reduce(
+    (acc, entity) => {
+      const market = addressToAsset(getAddress(entity.market)) as SupportedAsset
+
+      if (!acc[market]) acc[market] = []
+      acc[market].push(entity)
+      return acc
+    },
+    {} as Record<SupportedAsset, typeof accountPositionProcesseds>,
+  )
+
+  const trades = Object.entries(entities).reduce(
+    (acc, [market, entities]) => {
+      const asset = market as SupportedAsset
+
+      const updates: TradeHistoryItem<typeof accountPositionProcesseds>[] = []
+
+      entities.forEach((entity) => {
+        if (entity.update) {
+          updates.push({
```

Comment:
We'll need to re-use the same logic that subPosition has for populating these fields since there's a bit extra logic required here